modules/series-service-impl/src/main/java/org/opencastproject/series/impl/SeriesServiceImpl.java

```diff
@@ -587,15 +595,14 @@ public Void xapply() throws Exception {
                       messageSender.sendObjectMessage(destinationId, MessageSender.DestinationType.Queue,
                               SeriesItem.updateProperty(id, property.getKey(), property.getValue()));
                     }
-                    if (((current[0] % responseInterval) == 0) || (current[0] == total)) {
-                      messageSender.sendObjectMessage(IndexProducer.RESPONSE_QUEUE, MessageSender.DestinationType.Queue,
-                              IndexRecreateObject.update(indexName, IndexRecreateObject.Service.Series, total,
-                                      current[0]));
-                    }
-                    current[0] += 1;
                     return null;
                   }
                 });
+        if ((current % responseInterval == 0) || (current == total)) {
```

Comment:
Hi @lkiesow, I might be wrong, but from the experience we had with the original index performance optimization patches, this won't work:

This log will just log the progress of building up a queue of index messages, but that doesn't mean that those index rebuilt messages have been processed. 
My guess is that this is the reason why you find the weird asynchronous index progress logging across all ElasticSearch repopulate methods.

Reply:
This is the exact same conversation we had with the workflow index rebuild: Your observation is true in the sense that the logging will only track the initialization of adding the new objects to the index. Nevertheless, your assumption that this behavior was previously different is wrong. What we achieve now it the same thing we previously achieved only with fewer messages sent over ActiveMQ. The reason is that the status messages were previously sent to a separate queue which was unrelated and in no way synchronized to the objects sent for re-indexing.

If you want me to look into getting proper logs, I can do that. A reasonable approach would be to do a synchronized index rebuild without sending out all the messages which would also likely speed up the rebuild quite a bit again. But that would be a separate patch entirely.

Reply:
Honestly, I don't understand why it did work before, but from observation of the asset manager index rebuilt: Before the logging change on our system, all events have showed up once index rebuilt log indicated 100%. After the logging change, it took 8 hours until that situation was reached.

-> The async log does definitely indicate progress correctly (for whatever reason), but the synchronous logging does definitely not.

Is it possible that messageSender.sendObjectMessage(IndexProducer.RESPONSE_QUEUE...) blocks the caller until reaching a specific state of the messageSender.sendObjectMessage(destinationId,...)?
That would be the only explanation why this works.

Reply:
> If you want me to look into getting proper logs, I can do that. A reasonable approach would be to do a synchronized index rebuild without sending out all the messages which would also likely speed up the rebuild quite a bit again. But that would be a separate patch entirely.

Not necessary as the log works correctly so far - just don't break it.

Be careful about the idea of synchronous index rebuilds: In contrary to the Solr workflow index rebuild which blocks startup of Opencast, the ElasticSearch index rebuilds run while Opencast is running.
Assuming ("hoping that") all ElasticSearch index updates come over those ActiveMQ queues, the repopulate methods should work fine when users cause index updates, too (since the queue ensures correct order of index updates). If synchronous and asynchronous updates would be mixed up, ... well, let's just hope that is not the case ;-)

Reply:
> Not necessary as the log works correctly so far - just don't break it.

@staubesv,  this is simply not true. The logs have been slower before since we were hammering ActiveMQ but it was never a representation of the rebuild, only a representation of the initialization. Just to humor you, I've logged the re-build (logging + updates) in the latest state of `develop` on my test system wich holds 1000 series: https://data.lkiesow.de/opencast/opencast-repopulateindex.log

In particular, have a look at when the finish message is logged and when the processing of the last message actually starts:
```
2018-02-26 15:16:48,505 | ERROR | (AbstractSearchIndex:212) - Updating service: 'Series' with 1000/1000 finished, 100% complete.
...
2018-02-26 15:17:59,615 | ERROR | (SeriesMessageReceiverImpl:66) - Series message of type `UpdateOptOut` received
```

This clearly shows that it does not at all represent the re-build.

>  […] the repopulate methods should work fine when users cause index updates, too 

Again, simply not true. As a user, I can make changes while the messages are generated and the re-build would overwrite my change in the index.

Reply:
Hmm... well, thanks for verifying that my hypothesis was wrong ;-) So this was always broken but becomes more obvious now since logging takes longer.

Just wondering what this mean for the "rebuilt all indices".... as far as I understand, the order of the index rebuilds is relevant, i.e. workflow index must be completed before asset index...

> Again, simply not true. As a user, I can make changes while the messages are generated and the re-build would overwrite my change in the index.

That is scary as mixing up synchronous and asynchronous updates implies that the order of updates is not defined... I kind of hoped that user changes would generate a index rebuild message, too.

Reply:
Hmm.... Once I find some time, I will have a closer look at https://github.com/opencast/opencast/pull/28 again. That one did rename some of the logging queues...